### ISSUE TITLE
Buggy scenario

### ISSUE THREAD

**Issue by khanh-nguyen** [jamesdbloom/mockserver#143]
I want to have a scenario that query to an end point returns `401, 401` followed by infinite `200`. Here're the expectations I stubbed in sequence.
```
{
    httpRequest: {
        method: 'GET',
        path: '/path/to/x'
    },
    httpResponse: {
        statusCode: 401
    },
    times: {
        remainingTimes: 2,
        unlimited: false
    }
}

{
    httpRequest: {
        method: 'GET',
        path: '/path/to/x'
    },
    httpResponse: {
        statusCode: 200
    },
    times: {
    		unlimited: true
		}    		
}
``` 
As the result, after getting 2 `401` and 1 `200`, I get `404`.
However, if I only stub this
```
{
    httpRequest: {
        method: 'GET',
        path: '/path/to/x'
    },
    httpResponse: {
        statusCode: 200
    },
    times: {
    		unlimited: true
		}    		
}
```
then, I'm able to get `200` infinitely.

**Comment by jamesdbloom** [comment]
I'll take a look and get back on this.

**Comment by jamesdbloom** [comment]
This may be caused by two separate features overlapping but I need to double check this and add some tests to cover this exact scenario

**Comment by jamesdbloom** [comment]
This should be fixed in the current 3.9.16-SNAPSHOT and will be released in the next couple of days

**Comment by khanh-nguyen** [comment]
Yes, it's working correctly now. You're awesome, @jamesdbloom. Thanks a lot.
